File: 0733-flood-fill/0733-flood-fill.java
class Solution {
    public int[][] floodFill(int[][] image, int sr, int sc, int color) {
        int[][] ans = new int[image.length][];
        for(int i = 0;i<image.length;i++){
            ans[i] = image[i].clone();      
        }
        int org = image[sr][sc];
        int[][] dir = {{0,1},{0,-1},{1,0},{-1,0}};
        dfs(ans,sr,sc,color,dir,org);
        return ans;
    }
    private void dfs(int[][] ans , int i , int j , int color,int[][] dir , int org){
        ans[i][j] = color;
        for(int[] d : dir){
            int r = i + d[0];
            int c = j + d[1];
            if(r >= 0 && r < ans.length && c >=0 && c < ans[0].length && ans[r][c] == org && ans[r][c] != color){
                dfs(ans,r,c,color,dir,org);
            }
        }
    }
}

File: 0733-flood-fill/0733-flood-fill.py
class Solution:
    def floodFill(self, image: List[List[int]], sr: int, sc: int, color: int) -> List[List[int]]:
        original_col = image[sr][sc]
        if original_col == color:
            return image
        rows , cols = len(image) , len(image[0])
        def dfs(row : int,col : int):
            if row<0 or row>=rows or col < 0 or col >= cols or image[row][col] != original_col: 
                return
            
            image[row][col] = color
            dfs(row-1,col)
            dfs(row+1,col )
            dfs(row,col-1)
            dfs(row,col+1)
        dfs(sr,sc)
        return image
        